spring-cloud-kubernetes-commons/src/main/java/org/springframework/cloud/kubernetes/commons/config/reload/ConfigurationChangeDetector.java

```diff
@@ -103,41 +84,22 @@ public boolean changed(List<? extends MapPropertySource> left, List<? extends Ma
 		return false;
 	}
 
-	/**
-	 * Finds one registered property source of the given type, logging a warning if
-	 * multiple property sources of that type are available.
-	 * @param <S> property source type
-	 * @param sourceClass class for which property sources will be searched for
-	 * @return matched property source
-	 */
-	protected <S extends PropertySource<?>> S findPropertySource(Class<S> sourceClass) {
-		List<S> sources = findPropertySources(sourceClass);
-		if (sources.size() == 0) {
-			return null;
-		}
-		if (sources.size() > 1) {
-			this.log.warn("Found more than one property source of type " + sourceClass);
-		}
-		return sources.get(0);
-	}
-
 	/**
 	 * @param <S> property source type
 	 * @param sourceClass class for which property sources will be found
 	 * @return finds all registered property sources of the given type
 	 */
 	public <S extends PropertySource<?>> List<S> findPropertySources(Class<S> sourceClass) {
-		List<S> managedSources = new LinkedList<>();
+		List<S> managedSources = new ArrayList<>();
 
-		LinkedList<PropertySource<?>> sources = toLinkedList(this.environment.getPropertySources());
```

Comment:
I did not understand the purpose of a LinkedList here, when an ArrayList does preserve the same order, as such small refactor here